zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt

```diff
@@ -38,26 +41,39 @@ class LeakedServicesTest {
     uncaughtExceptionHandler.tearDown()
   }
 
-  @Test fun jvmLeaksService() {
+  @Test fun jvmLeaksService(): Unit = runBlocking(dispatcher) {
     zipline.quickJs.evaluate("testing.app.cash.zipline.testing.prepareJsBridges()")
     allocateAndLeakService()
     awaitGarbageCollection()
-    triggerLeakDetection()
+    triggerJvmLeakDetection()
     val name = "helloService"
     assertThat(eventListener.take()).isEqualTo("takeService $name")
     assertThat(eventListener.take()).isEqualTo("serviceLeaked($name)")
   }
 
-  /** Just attempting to take a service causes Zipline to process leaked services. */
-  private fun triggerLeakDetection() {
-    try {
-      zipline.take<EchoService>("noSuchService")
-    } catch (ignored: Exception) {
+  @Test fun jsLeaksService(): Unit = runBlocking(dispatcher) {
+    val supService = object : EchoService {
+      override fun echo(request: EchoRequest): EchoResponse = error("unexpected call")
     }
+
+    val name = "supService"
+    zipline.bind<EchoService>(name, supService)
+    zipline.quickJs.evaluate("testing.app.cash.zipline.testing.allocateAndLeakService()")
+    zipline.quickJs.evaluate("testing.app.cash.zipline.testing.triggerLeakDetection()")
+    assertThat(eventListener.take()).isEqualTo("bindService $name")
+    assertThat(eventListener.take()).isEqualTo("serviceLeaked($name)")
   }
 
   /** Use a separate method so there's no hidden reference remaining on the stack. */
   private fun allocateAndLeakService() {
-    zipline.take<EchoService>("helloService")
+    zipline.take<EchoService>("helloService") // Deliberately not closed for testing.
+  }
+
+  /** Just attempting to take a service causes Zipline to process leaked services. */
+  private fun triggerJvmLeakDetection() {
+    try {
+      zipline.take<EchoService>("noSuchService")
```

Comment:
Is the throwing random? If we're expecting it to throw, we could follow a `fail()` method

Reply:
Should use `assertFailsWith<Exception> { ... }`

Reply:
Fixed.